json-logic/schema.ts

```diff
@@ -0,0 +1,146 @@
+// LogicExpression refers to the JSONLogic standard.
+import { LogicExpression } from "./logic";
+
+/*
+* This module defines a schema for questionaire and result calulation logic.
+*/
+
+/**
+ * Type of a question, essentially defines the result type.
+ */
+export enum QuestionType {
+	Select = "Select",
+	Multiselect = "Multiselect",
+	Number = "Number",
+	Boolean = "Boolean",
+	Date = "Date",
+	Text = "Text"
+}
+
+/**
+ * Option for multi-select questions.
+ */
+export interface IOption {
+	/**
+	 * Human-Readable answer, can be localized.
+	 */
+	text: string,
+	/**
+	 * Value used for evaluating logic expressions.
+	 */
+	value: string;
+}
+
+/**
+ * Represents a question.
+ */
+export interface IQuestion {
+	/**
+	 * Unique id for referring this question in logic expressions.
+	 */
+	id: string
+	/**
+	 * Type of the question.
+	 */
+	type: QuestionType,
+	/**
+	 * Human-readable question text, can be localized.
+	 */
+	text: string,
+	/**
+	 * Answer options for Select/Multiselect questions.
+	 */
+	options?: IOption[],
+	/**
+	 * Boolean indicating whether the question is optional or not.
+	 */
+	optional?: boolean
+	/**
+	 * Logic expression to decide whether the question should be displayed or not.
+	 */
+	skipIf?: LogicExpression
```

Comment:
Should we call it `enableWhen` so the logical not of `skipIf`? This would be compliant with FHIR.

Reply:
I am voting for this. but it should be `x_enableWhen` so it is clear its not the standard. 
Most of the statements in `skipIf` are inverted at the moment. `enableWhen` does not need the inversion.

Reply:
All questions are enabled by default, and skipping is the corner case. Hence the field should be optional and deal with skipping, not enabling. 

I also oppose marking things not in FHIR with `x_`. A good solution for supporting FHIR when it's needed is providing a mapping when it's needed (as we discussed multiple times). We only need to take care of structural compatibility, field names can simply be remapped.

Reply:
Well if `enableWhen` is empty it is treated as true. But I get your point. 
With the `x_` I just wanted to mark the difference, if we use a different name entirely there is no chance to collide with already defined attributes.

Reply:
I lean towards inverting the logic and using some optional field like enableWhen, since I think it is easier to understand. Default as Daniel said is then that the question is shown. We could use the name "shownIf" or "visibleIf".